WordPress/Classes/Services/SiteAssembly.swift

```diff
@@ -47,41 +50,43 @@ final class MockSiteAssemblyService: NSObject, SiteAssemblyService {
     private(set) var currentStatus: SiteAssemblyStatus = .idle {
         didSet {
             DispatchQueue.main.async { [weak self] in
-                guard let strongSelf = self else {
+                guard let self = self else {
                     return
                 }
-                strongSelf.statusChangeHandler?(strongSelf.currentStatus)
+                self.statusChangeHandler?(self.currentStatus)
```

Comment:
This is all very new to me and I am not familiar with how logging is done but would it make sense to add a logging here if the newValue is .failed or just rely on the caller statusChangeHandler to handle?

Reply:
This particular invocation is part of the mocked service, so I opted not to add additional logging here. 

That said, the comment is still valid for `SiteAssemblyService.swift:17`. In that case, we have some error logging in its caller (i.e., `SiteAssemblyWizardContent.attemptSiteCreation()`). This will be enhanced further via #10335.

Reply:
👍